crud/src/main/java/com/redhat/lightblue/mediator/DefaultMetadataResolver.java

```diff
@@ -180,25 +191,81 @@ private void addFieldRoles(Set<String> roles, FieldTreeNode node) {
         }
     }
 
-    private EntityMetadata getMetadata(String entityName,
-                                       String version) {
-        EntityMetadata emd = metadataMap.get(entityName);
-        if (emd != null) {
-            if (!emd.getVersion().getValue().equals(version)) {
-                throw Error.get(CrudConstants.ERR_METADATA_APPEARS_TWICE, entityName + ":" + version
-                        + " and " + emd.getVersion().getValue());
+    private void initMetadataMap(EntityMetadata emd, String entityName, String rootRequestVersion) {
+        // store root entity metadata
+        metadataMap.put(entityName, emd);
+        Set<String> deferred = new HashSet<>();
+        FieldCursor cursor = emd.getFieldCursor();
+        while (cursor.next()) {
+            FieldTreeNode node = cursor.getCurrentNode();
+            if (node instanceof ReferenceField) {
+                ReferenceField ref = (ReferenceField) node;
+                String name = ref.getEntityName();
+                String ver = ref.getVersionValue();
+                EntityMetadata refMd;
+                if (ver == emd.getEntityInfo().getDefaultVersion()) {
```

Comment:
if association is using default version, ver=null. So, this comparison is wrong. if(ver==null), then add this to deferred loads, so you can load it later.